interfaces/repo.go

```diff
@@ -449,28 +449,73 @@ func (r *Repository) PlugConnections(snapName, plugName string) []*Slot {
 	return result
 }
 
-// Support for sort.Interface
-
-type byPlugSnapAndName []*Plug
+// SlotConnections returns all of the plugs that are connected a given slot.
+func (r *Repository) SlotConnections(snapName, slotName string) []*Plug {
+	r.m.Lock()
+	defer r.m.Unlock()
 
-func (c byPlugSnapAndName) Len() int      { return len(c) }
-func (c byPlugSnapAndName) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
-func (c byPlugSnapAndName) Less(i, j int) bool {
-	if c[i].Snap != c[j].Snap {
-		return c[i].Snap < c[j].Snap
+	slot := r.slots[snapName][slotName]
+	if slot == nil {
+		return nil
 	}
-	return c[i].Plug < c[j].Plug
+	var result []*Plug
+	for plug := range r.slotPlugs[slot] {
+		result = append(result, plug)
+	}
+	sort.Sort(byPlugSnapAndName(result))
+	return result
 }
 
-type bySlotSnapAndName []*Slot
+// InterfaceConnections returns object holding a lists of all the plugs and slots and their connections.
```

Comment:
`// Interfaces returns all plugs, slots, and their connections.`

I wasn't considering having that and SlotRef/PlugRef in the internal APIs, but maybe that's a good idea.